brand/utils/harvest_data.py

```diff
@@ -0,0 +1,47 @@
+from datetime import datetime, timedelta
+from urllib.parse import quote
+
+from django.conf import settings
+
+import requests
+
+
+def fetch_harvest_data(brand_tag, brand_url="", brand_country="", brand_name=""):
+    base_url = "https://bank.green/harvest"
+
+    params = {
+        "bankTag": brand_tag,
+        "bankUrl": brand_url,
+        "bankCountry": brand_country,
+        "bankName": brand_name,
+    }
+
+    # URL encode the parameters
+    encoded_params = "&".join(f"{k}={quote(v)}" for k, v in params.items() if v)
+
+    url = f"{base_url}?{encoded_params}"
+
+    response = requests.get(url, headers={"Authorization": f"Token {settings.HARVEST_API_TOKEN}"})
+
+    if response.status_code == 200:
+        return response.json()
+    else:
+        return None
+
+
+def update_commentary_feature_data(commentary, overwrite=False):
```

Comment:
This looks like you're getting a bit ahead of yourself since it's out of scope for the ticket. Fine to keep it in, though you should probably either make it async or call `fetch_harvest_data` asynchronously with a `ThreadPoolExecutor`. I'm not that good at async stuff, TBH, so a chatbot may be helpful to you here.

If you want to keep it in as is though, that's also fine. It's not being used until the next ticket anyway.

Reply:
ooh, I see the 90 day TTL thing here. Yeah, that make sense.

Reply:
For simplicity and due to my current time constraint i have right now, keeping it as is. Think we will come back to it anyway.